ui/src/frontend/track_group_panel.ts

```diff
@@ -227,17 +227,35 @@ export class TrackGroupPanel extends Panel<Attrs> {
     if (!(e.target instanceof HTMLElement)) return;
     const dataTransfer = e.dataTransfer;
     if (dataTransfer === null) return;
-    if (!dataTransfer.types.includes('perfetto/track')) return;
+    const dataType = dataTransfer.types.find((dataType)=>{
+      if (dataType.startsWith('perfetto/track/')) {
+        return true;
+      }
+      return false;
+    });
+    if (!dataType) return;
+    const trackLikeId = dataType.split('/').pop();
+    if (!trackLikeId) return;
     e.stopPropagation();
     dataTransfer.dropEffect = 'move';
     e.preventDefault();
 
-    // Apply some hysteresis to the drop logic so that the lightened border
-    // changes only when we get close enough to the border.
-    if (e.offsetY < e.target.scrollHeight / 3) {
-      this.dropping = 'before';
-    } else if (e.offsetY > e.target.scrollHeight / 3 * 2) {
-      this.dropping = 'after';
+    // Test if id has same parent as current
+    // If no do not set this.dropping
+    let trackLike : TrackState | TrackGroupState =
+      globals.state.trackGroups[trackLikeId];
+    if (!trackLike) {
+      trackLike = globals.state.tracks[trackLikeId];
+    }
```

Comment:
It's arguably more clear about the intent to use the modern nullish coalescing operator. Maybe not.
```suggestion
    let trackLike : TrackState | TrackGroupState =
      globals.state.trackGroups[trackLikeId] ?? globals.state.tracks[trackLikeId];
```